### PR TITLE
Add autoscaling documentation

### PR DESCRIPTION
**Pull request by guilhermocc**
## What ❓ 
Add autoscaling feature documentation section.

### CODE REVIEW

File: docs/Autoscaling.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+|         100        |           30          |     0.5     |          60          |   Scale Down: -40   |
+|         50         |           40          |     0.3     |          58          |    Scale Up: +8     |
+|         50         |           35          |     0.3     |          50          |    Do Nothing: 0    |
+|         50         |           10          |     0.3     |          15          |   Scale Down: -35   |
+|         10         |           5           |     0.9     |          50          |    Scale Up: +40    |
+|         10         |           1           |     0.9     |          10          |    Do Nothing: 0    |
+|         10         |           1           |     0.8     |           5          |   Scale Down: -5    |
+|          5         |           5           |     0.1     |           6          |    Scale Up: +1     |
+|          1         |           1           |     0.3     |           2          |    Scale Up: +1     |
+|          2         |           2           |     0.9     |          20          |    Scale Up: +18    |
```
Comment (guilhermocc) [inline docs/Autoscaling.md:150]: Preview:
image

File: docs/Autoscaling.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+```mermaid
+  flowchart TD
+    finish((End))
+    add_rooms_operation(Enqueue add rooms operation)
+    remove_rooms_operation(Enqueue remove rooms)
+    use_rooms_replicas(Use rooms replicas to calculate the desired number of rooms)
+    autoscaling_enabled{Autoscaling configured and enabled?}
+    decide_operation{Compare current number of rooms with the desired amount.}
+    use_autoscaling[Use Autoscaling policy to calculate the desired number of rooms corced in min max range]
```
Comment (caiordjesus) [inline docs/Autoscaling.md:18]:
```suggestion
    use_autoscaling[Use Autoscaling policy to calculate the desired number of rooms coerced in min-max range]
```

File: docs/Autoscaling.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+Currently, the sync interval is configured by environment variable `MAESTRO_WORKERS_OPERATIONEXECUTION_HEALTHCONTROLLERINTERVAL`.
+
+> By default, the scheduler does not have autoscaling configured.
+
+### How to configure and enable autoscaling
+To get autoscaling working in your scheduler, firstly you need to configure an autoscaling policy and enable it, this autoscaling
+configuration resides in the root of the scheduler structure itself.
+
+Yaml version
+```yaml
```
Comment (caiordjesus) [inline docs/Autoscaling.md:37]: WDYT about adding these guys inside a dropdown like in Scheduler.md?

File: docs/Autoscaling.md [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+        ...
+        // Will vary according to the policy type.
+      }
+    }
+  }
+}
+```
+
+- **enabled** [boolean]: A value that can be true or false, indicating if the autoscaling feature is enabled/disabled for the given scheduler. Default: false.
+- **min** [integer]: Minimum number of rooms the scheduler should have, it must be greater than zero.
```
Comment (caiordjesus) [inline docs/Autoscaling.md:74]:
```suggestion
- **min** [integer]: Minimum number of rooms the scheduler should have, it must be greater than zero. For zero value, disable autoscaling and set "roomsReplicas" to 0.
```

Comment (caiordjesus) [reply docs/Autoscaling.md:74]: Trust me, this information is very relevant for people rolling out the docs trying to understand how to have 0 replicas

File: docs/Autoscaling.md [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+So basically Maestro will constantly try to maintain a certain percentage of rooms in **ready** state, by looking at the
+actual room occupancy rate (number of rooms in **occupied** state).
+
+#### Room Occupancy Policy Parameters
+- **readyTarget** [float]: The percentage (in decimal value) of rooms that Maestro should try to keep in **ready** state, must be a value between 0.1 and 0.9.
+
+#### Example
+
+Yaml version
+```yaml
```
Comment (caiordjesus) [inline docs/Autoscaling.md:102]: Same dropdown comment

**Review by caiordjesus — CHANGES_REQUESTED** [review]
Update Scheduler.md to have autoscaling link, add example to example scheduler on that file also

**Review by caiordjesus — APPROVED** [review]

**Pull request merged**